dbus/methods_test.go

```diff
@@ -523,6 +732,107 @@ func TestStartStopTransientScope(t *testing.T) {
 	//     int sd_pid_get_unit(pid_t pid, char **session)
 }
 
+// Ensure that basic unit gets killed by SIGTERM
+func TestKillUnit(t *testing.T) {
+	target := "start-stop.service"
+	conn := setupConn(t)
+
+	setupUnit(target, conn, t)
+	linkUnit(target, conn, t)
+
+	// Start the unit
+	reschan := make(chan string)
+	_, err := conn.StartUnit(target, "replace", reschan)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	job := <-reschan
+	if job != "done" {
+		t.Fatal("Job is not done:", job)
+	}
+
+	// send SIGTERM
+	jobDone := false
+	maxAttempts := 3
+	for i := 0; i < maxAttempts; i++ {
+		conn.KillUnit(target, int32(syscall.SIGTERM))
+
+		units, err := conn.ListUnits()
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		unit := getUnitStatus(units, target)
```

Comment:
@lucab Hmm, that sounds like a good idea. Will try it. I also think such a polling is ugly.